lib/kennel/models/screen.rb

```diff
@@ -147,6 +148,37 @@ def widget_defaults(type)
 
         basics.merge(custom)
       end
+
+      def resolve_uptime_widget_link(widget, id_map)
```

Comment:
turn this into a single `resolve_link widget, key, type, id_map` and then call `resolve_link widget, [:monitor, :id], Integer, id_map`